scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-1x.yaml

```diff
@@ -170,6 +170,39 @@ QFN-16-1EP_3x3mm_P0.5mm_EP1.45x1.45mm:
   #suffix: '_Pad{pad_x:.2f}x{pad_y:.2f}mm_HandSolder'
   #include_suffix_in_3dpath: 'False'
 
+QFN-16-1EP_3x3mm_P0.5mm_EP1.6x1.6mm:
+  device_type: 'QFN'
+  ipc_class: 'qfn'
+  size_source: 'https://www.diodes.com/assets/Datasheets/PI6C5946002.pdf'
+  body_size_x:
+    minimum: 2.9
+    nominal: 3.0
+    maximum: 3.1
+  body_size_y:
+    minimum: 2.9
+    nominal: 3.0
+    maximum: 3.1
+  overall_height:
+    minimum: 0.7
+    nominal: 0.75
+    maximum: 0.8
+
+  lead_width:
+    minimum: 0.18
+    maximum: 0.3
+  lead_len:
+    minimum: 0.3
+    maximum: 0.5
+
+  EP_size_x: 1.6
+  EP_size_y: 1.6
```

Comment:
Use package pad dimensions and tolerances as given in the datasheet and then add EP_size_[xy]_overwrite to get a 1.6x1.6mm FP pad.